tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/entity/FixResult.java

```diff
@@ -27,17 +29,25 @@
   /** The position at the query where the error occurs. */
   Position errorPosition;
 
+  /** Is query fixer confident about this fixing */
+  Boolean isConfident;
```

Comment:
why we want this flag? What if we are not confident？
Maybe give confidence level instead of boolean?

Reply:
The confident flag is used when the program is in user-assistance mode. Usually, the program only asks users to select an option if multiple ones exist. However, if an approach is marked not confident, then the program will ask users to confirm the option even if only one option exists.

Reply:
ok. How do we know if we are confident or not?

Reply:
Currently, it is determined by developers. For example, fixing the unrecognized column is confident, because the column to replace is the most similar column to the incorrect one. 

However, fixing some syntax error could be not confident. One case is "Expect End of Input but got others". Most unparsable query will result in this error, but this error message contains no useful information. Thus, there is no way to fix this error in a confident way. In this situation, the query fixer will mark this fix as "not confident".